tests/publiccloud/aistack_basic.pm

```diff
@@ -0,0 +1,198 @@
+    my $docker_user_name = 'yarunachalam@suse.com';
+    my $repo_url = 'github.com/SUSE/private-ai-charts';
+    # my $repo_url = get_var('HELM_CHARTS');
+    my $git_token = get_var('_SECRET_SSH');
+
+    # config kubectl
+    assert_script_run("mkdir -p ~/.kube");
+    assert_script_run("sudo cp /etc/rancher/rke2/rke2.yaml ~/.kube/config");
+    assert_script_run("kubectl config get-contexts");
+    assert_script_run("kubectl config use-context default");
+    assert_script_run("kubectl config view");
+
+    # Add cert-manager repo,install
+    assert_script_run("helm repo add $cert_repo");
+    assert_script_run("helm repo update");
+    assert_script_run("helm install cert-manager jetstack/cert-manager --namespace cert-manager --create-namespace --version v1.15.2 --set crds.enabled=true", timeout => 120);
+
+    # Add Ingress Controller to open-webui endpoint
+    assert_script_run("helm repo add $ingress_repo");
+    assert_script_run("helm repo update");
+    assert_script_run("helm search repo ingress-nginx -l");
+
+    sleep 180;
+    assert_script_run("helm list --all-namespaces");
+    assert_script_run("kubectl get pods --all-namespaces");
+
+    # Access to Application collection registery
+    # Get docker username and password
+    assert_script_run("kubectl create ns $namespace");
+    assert_script_run("kubectl create secret docker-registry application-collection --docker-server=dp.apps.rancher.io --docker-username='$docker_user_name' --docker-password='$SECRET_application_collection' -n $namespace", timeout => 120);
+
+    # Install private-ai-stack
+    my $gitlab_clone_url = 'https://git:' . $git_token . '@' . $repo_url;
+    assert_script_run("git clone $gitlab_clone_url");
+    assert_script_run("curl -o $vf_name $ai_chart_repo", timeout => 120);
+    assert_script_run("kubectl apply -f https://raw.githubusercontent.com/rancher/local-path-provisioner/v0.0.28/deploy/local-path-storage.yaml", timeout => 120);
+    assert_script_run("helm upgrade --install suse-private-ai private-ai-charts --namespace $namespace --create-namespace --values $vf_name", timeout => 300);
+    assert_script_run("kubectl get all --namespace $namespace");
+    sleep 120;
+    # Check pod status and log for successful
+    my $max_retries = 15;
+    my $sleep_interval = 10;
+    my @out = split(' ', script_output("kubectl get pods --namespace $namespace -o custom-columns=':metadata.name'"));
+    record_info("Pod names", join(" ", @out));
+    POD_LOOP: foreach my $pod (@out) {
+        my $counter = 0;
+        my $start_time = time();
+        while ($counter++ < $max_retries) {
+            my $status = script_output("kubectl get pod $pod -n $namespace -o=jsonpath='{.status.phase}'", proceed_on_failure => 1);
+            my $logs = script_output("kubectl logs $pod -n $namespace", proceed_on_failure => 1);
+            if ($status eq 'Running') {
+                record_info("$pod is running. ");
+                next POD_LOOP;
```

Comment:
No need of a label for `next`, by default it directly goes to inner `foreach`: you can remove labels unless nested loops to jump.

Reply:
when pod is running or failure adn other status it should jump to next job..Unless it is pending..will loop thru the while loop ..tested 2-3 times as expected logic works.

Reply:
ok, but non need for a label `POD_LOOP`, that you can remove from everywhere.

Reply:
ok, but to skip to next pod you simply go to the end of while with `last`, then naturally proceed towards the end of for-loop, that goes on.
I mean, in that case you need a `last`, not a `next`.

Reply:
Yes, the loop-breaking mechanism does work and is rare cases useful, when 2 or more nested loops you want to jump from inner to outer; but it is discouraged in good structured programming guidelines, for many operational motivations.